crates/solidity/src/solc/solc_compiler.rs

```diff
@@ -56,42 +64,58 @@ impl Compiler for SolcCompiler {
         let mut command = std::process::Command::new(self.executable.as_str());
         command.stdin(std::process::Stdio::piped());
         command.stdout(std::process::Stdio::piped());
-        command.arg("--standard-json");
 
-        if let Some(base_path) = base_path {
-            command.arg("--base-path");
-            command.arg(base_path);
-        }
-        for include_path in include_paths.into_iter() {
-            command.arg("--include-path");
-            command.arg(include_path);
+        if let Some(base_path) = &base_path {
+            if SUPPORTS_BASE_PATH.matches(&version.default) {
+                command.arg("--base-path").arg(base_path);
+
+                if SUPPORTS_INCLUDE_PATH.matches(&version.default) {
+                    // `--base-path` and `--include-path` conflict if set to the same path, so
+                    // as a precaution, we ensure here that the `--base-path` is not also used
+                    // for `--include-path`
+                    for path in include_paths.iter().filter(|p| *p != base_path) {
+                        command.arg("--include-path").arg(Path::new(path));
+                    }
+                }
+            }
```

Comment:
I think instead of silently swallong the flag it should error out here. The user passing invalid solc arguments hints at a problem. We should not silenty ignore but treat it as an error instead.

Reply:
But remember this will be used by various tools with different solc versions so would we then need to abort the whole process of compilation

Reply:
This is exactly my point. Invoking the compiler with invalid configuration should abort the compilation process instead of silently ignoring it.